scapy/arch/bpf/supersocket.py

```diff
@@ -354,7 +354,10 @@ def recv(self, x=BPF_BUFFER_LENGTH):
         """Receive on layer 3"""
         r = SuperSocket.recv(self, x)
         if r:
-            return r.payload
+            ts = r.time
+            r = r.payload
+            r.time = ts
```

Comment:
No need for a temporary variable here or another variable assignment:
```suggestion
            r.payload.time = r.time
            return r.payload
```